src/content/blog/2020-12-07-roadmap-2021.md

```diff
@@ -0,0 +1,238 @@
+This is something we want to look into for webpack, but also for terser, which is used by webpack as minimizer by default.
+
+### `exports`/`imports` package.json field
+
+Node.js 14 has added supported for the `exports` field in package.json to allow to define entrypoints of a package.
+Webpack 5 followed this, and even added additional conditions like `production/development`.
+
+Shortly after that Node.js made further additionals to that, e. g. they also added an `imports` field for private imports.
+
+That's something we also want to add.
+
+### Improve CommonJS analysis
+
+While ESM is the future, there are still a lot CommonJS packages in npm and in use.
+Webpack 5 added analysis for CommonJS modules to allow Tree Shaking for most of these modules.
+
+But we can do more. While many exporting patterns are supported, only a few importing patterns are supported.
+We want to add support for more patterns to allow more optimizations for CommonJS modules too.
+
+### Hot Module Replacement for Module Federation
+
+Webpack 5 added a new feature called "Module Federation" which allows to integrate multiple builds together at runtime.
+Currently Hot Module Replacement (HMR) supports only a single build at a time and updates can't bubble between build.
+We want to improve here and allow HMR updates to bubble between different builds, which would improve developing federation applications.
+
+### Hinting system
+
+Currently webpack displays warnings and errors to the user.
+During build there are quite a few cases where we could tell the user something, like potential footguns or optimization opportunities, but they doesn't fit into warnings or errors and we don't want to spam the output with all this information.
+So we want to add another category: Hints.
+We want to collect all hints during builds (plugins could also emit some), but only display a limited number of them in the output (by default only one).
+This should lead to some kind of "Did you know" experience for the user.
+
+### Multi-Threading
+
+While Persistent Caching allows to make cached build "blazing" fast, initial builds without Persistent Cache have still a bit room for improvement.
+Javascript execution in Node.js is single-threaded by default, but recent additions allow to use `worker_threads`, which is an API similar to WebWorkers.
+
+This can be used distribute work across all CPUs. There has already been some preparations in webpack 5 for that: e. g. serializing of internal datastructures is possible and the work queues support plugins. But some parts of that are still unclear and require experimentation.
+
+This is in our voting list for a while, but not many have voted on that. Is this really something people need?
+
+### WebAssembly
+
+Currently WebAssembly is experimental and not enabled by default.
+Once the proposal reaches Stage 4 we can enable it by default.
+
+This might also lead to wider adoption on WebAssembly in the ecosystem.
```

Comment:
```suggestion
Currently, WebAssembly is experimental and not enabled by default.
Once the proposal reaches Stage 4 we can enable it by default.

This might also lead to a wider adoption of WebAssembly in the ecosystem.
```